packages/widgets/src/blocks/widget-group/edit.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	useBlockProps,
+	BlockIcon,
+	ButtonBlockAppender,
+	InnerBlocks,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { Placeholder, TextControl } from '@wordpress/components';
+import { group as groupIcon } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { getBlockType } from '@wordpress/blocks';
+
+export default function Edit( props ) {
+	const { clientId, isSelected } = props;
+	const { innerBlocks } = useSelect( ( select ) =>
+		select( blockEditorStore ).getBlock( clientId )
+	);
+
+	let content;
+	if ( innerBlocks.length === 0 ) {
+		content = <PlaceholderContent { ...props } />;
+	} else if ( isSelected ) {
+		content = <EditFormContent { ...props } innerBlocks={ innerBlocks } />;
+	} else {
+		content = <PreviewContent { ...props } innerBlocks={ innerBlocks } />;
+	}
+
+	return (
+		<div { ...useBlockProps( { className: 'widget' } ) }>{ content }</div>
+	);
+}
+
+function PlaceholderContent( { clientId } ) {
+	return (
+		<>
+			<Placeholder
+				className="wp-block-widget-group__placeholder"
+				icon={ <BlockIcon icon={ groupIcon } /> }
+				label={ __( 'Widget Group' ) }
+			>
+				<ButtonBlockAppender rootClientId={ clientId } />
+			</Placeholder>
+			<InnerBlocks renderAppender={ false } />
+		</>
+	);
+}
+
+function EditFormContent( { attributes, setAttributes, innerBlocks } ) {
+	return (
+		<div className="wp-block-widget-group__edit-form">
+			<h2 className="wp-block-widget-group__edit-form-title">
+				{ __( 'Widget Group' ) }
+			</h2>
+			<TextControl
+				label={ __( 'Title' ) }
+				placeholder={ getDefaultTitle( innerBlocks ) }
+				value={ attributes.title }
```

Comment:
This is undefined when we first enter the form, so there's a console error popping up when we focus the input:

`Warning: A component is changing an uncontrolled input to be controlled. This is likely caused by the value changing from undefined to a defined value, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components`

Can we set it to some default value like empty string or something?